### PR TITLE
Match pyOpenSSL and service_identity to Twisted

### LINKED ISSUE #5621: Support for Twisted 22.8.0
Twisted 22.8.0 was released recently, and it says:

> Twisted now works with Cryptography versions 37 and above, and as a result, its minimum TLS protocol version has been upgraded to TLSv1.2.

Consequently, tests on some envs, including 3.8, now fail because they install older cryptography.

### PR DESCRIPTION
**Pull request by gliptak**
Resolves #5621

Signed-off-by: Gábor Lipták <gliptak@gmail.com>

Correcting https://github.com/scrapy/scrapy/actions/runs/3052958225/jobs/4922962575

```
    from twisted.internet import error, tcp, udp
.tox/py/lib/python3.8/site-packages/twisted/internet/tcp.py:38: in <module>
    from twisted.internet._newtls import (
.tox/py/lib/python3.8/site-packages/twisted/internet/_newtls.py:18: in <module>
    from twisted.protocols.tls import TLSMemoryBIOFactory, TLSMemoryBIOProtocol
.tox/py/lib/python3.8/site-packages/twisted/protocols/tls.py:45: in <module>
    from twisted.internet._sslverify import _setAcceptableProtocols
.tox/py/lib/python3.8/site-packages/twisted/internet/_sslverify.py:1828: in <module>
    defaultCiphers = OpenSSLAcceptableCiphers.fromOpenSSLCipherString(
.tox/py/lib/python3.8/site-packages/twisted/internet/_sslverify.py:1807: in fromOpenSSLCipherString
    SSL.TLS_METHOD,
E   AttributeError: module 'OpenSSL.SSL' has no attribute 'TLS_METHOD'
```

### CODE REVIEW

File: tests/test_crawler.py [hunk 1]
```diff
@@ -327,7 +327,7 @@ def test_reactor_default(self):
 
     def test_reactor_default_twisted_reactor_select(self):
         log = self.run_script('reactor_default_twisted_reactor_select.py')
-        if platform.system() == 'Windows':
+        if platform.system() in ['Windows', 'Darwin']:
```
Comment (Gallaecio) [inline tests/test_crawler.py:330]: Hang on in there, Linux!

**Review by Gallaecio — APPROVED** [review]
Great work!

@wRAR I would like to hear from you if you think there is some other approach we should explore here. But I did not realize upgrading some dependencies would do the job, and I am more than fine with this approach.

**Comment by Gallaecio** [comment]
> Great work!
> 
> @wRAR I would like to hear from you if you think there is some other approach we should explore here. But I did not realize upgrading some dependencies would do the job, and I am more than fine with this approach.

Nevermind, I believe you were already aware of this, we were just deciding whether the side effect (not supporting TLS ≤1.1) was OK.

**Comment by kmike** [comment]
Thanks @gliptak!

**Comment by kmike** [comment]
It seems tests are failing now with a different error, weird! See e.g. https://github.com/scrapy/scrapy/actions/runs/3123148006/jobs/5065580786 - there's a lot of errors like this:

> ___________________ ERROR collecting tests/test_webclient.py ___________________
tests/test_webclient.py:23: in <module>
    from scrapy.core.downloader.contextfactory import ScrapyClientContextFactory
scrapy/core/downloader/contextfactory.py:11: in <module>
    from scrapy.core.downloader.tls import DEFAULT_CIPHERS, openssl_methods, ScrapyClientTLSOptions
scrapy/core/downloader/tls.py:23: in <module>
    METHOD_SSLv3: SSL.SSLv3_METHOD,                     # SSL 3 (NOT recommended)
E   AttributeError: module 'OpenSSL.SSL' has no attribute 'SSLv3_METHOD'

**Comment by Gallaecio** [comment]
> weird!

It is your fault, you should have merged a few minutes earlier! https://pypi.org/project/pyOpenSSL/#history :facepalm:

**Pull request merged**